src/amo/components/Collection/index.js

```diff
@@ -0,0 +1,202 @@
+import Link from 'amo/components/Link';
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import Paginate from 'core/components/Paginate';
+import Card from 'ui/components/Card';
+import LoadingText from 'ui/components/LoadingText';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { ReactRouterLocation } from 'core/types/router';
+
+import './styles.scss';
+
+
+type Props = {|
+  collection: CollectionType | null,
+  dispatch: Function,
+  errorHandler: ErrorHandlerType,
+  i18n: Object,
+  loading: boolean,
+  location: ReactRouterLocation,
+  params: {|
+    slug: string,
+    user: string,
+  |},
+|};
+
+export class CollectionBase extends React.Component {
+  componentWillMount() {
+    this.loadDataIfNeeded();
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    this.loadDataIfNeeded(nextProps);
+  }
+
+  loadDataIfNeeded(nextProps?: Props) {
+    const { collection, errorHandler, loading, params } = {
+      ...this.props,
+      ...nextProps,
+    };
+
+    if (errorHandler.hasError()) {
+      log.warn('Not loading data because of an error.');
+      return;
+    }
+
+    if (loading) {
+      return;
+    }
+
+    let collectionChanged = false;
+    let addonsPageChanged = false;
+    let location = this.props.location;
+
+    if (nextProps && nextProps.location) {
+      const nextLocation = nextProps.location;
+
+      if (location.pathname !== nextLocation.pathname) {
+        collectionChanged = true;
+        location = nextLocation;
+      }
+
+      if (location.query.page !== nextLocation.query.page) {
+        addonsPageChanged = true;
+        location = nextLocation;
+      }
+    }
+
+    if (!collection || collectionChanged) {
+      this.props.dispatch(fetchCollection({
+        errorHandlerId: errorHandler.id,
+        page: parsePage(location.query.page),
+        slug: params.slug,
+        user: params.user,
+      }));
+
+      return;
+    }
+
+    if (collection && addonsPageChanged) {
+      this.props.dispatch(fetchCollectionPage({
+        errorHandlerId: errorHandler.id,
+        page: parsePage(location.query.page),
+        slug: params.slug,
+        user: params.user,
+      }));
+    }
+  }
+
+  props: Props;
+
+  url() {
+    const { params } = this.props;
+
+    return `/collections/${params.user}/${params.slug}/`;
+  }
+
+  renderCollection() {
+    const { collection, i18n, loading, location } = this.props;
+
+    const addons = collection ? collection.addons : [];
+
+    return (
+      <div className="Collection-wrapper">
+        <Card className="Collection-detail">
+          <h1 className="Collection-title">
+            {collection ? collection.name : <LoadingText />}
+          </h1>
+          <p className="Collection-description">
+            {collection ? collection.description : <LoadingText />}
+          </p>
+          <div className="Collection-metadata">
+            <dl className="Collection-number-of-addons">
+              <dd>
```

Comment:
`<dd>` first?! 😱

Seriously though I see why, I think that makes sense. It's just odd to see 😄